modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java

```diff
@@ -521,7 +523,7 @@ private void processMultiplatformLibrary(final String infrastructureFolder) {
         typeMapping.put("object", "kotlin.String");  // kotlin.Any not serializable
 
         // multiplatform import mapping
-        importMapping.put("BigDecimal", "kotlin.Double");
+        importMapping.put("BigDecimal", "Double");
         importMapping.put("UUID", "kotlin.String");
```

Comment:
All of these types should also probably be mapped in a better way if possible somehow.
Maybe with custom classes that can be converted into local types then (if available)

Reply:
Loss of data should be avoided, but if it's possible to use some wrapper type, instead of the string, that can be converted to the equivalent types of that platform (like `LocalDatetime` or some of the other APIs for the jvm, `Date` for js and probably something for iOS).
Some of them, like URI, might already have primitives in kotlin std or ktor.

Reply:
What was the reason for changing `kotlin.Double` to `Double` here? I'd have to test, but iirc kotlin client would cause this to become a java.lang.Double (or something similar) and wouldn't honor kotlin's nullable functionality.

Reply:
Kotlin uses the `kotlin.*` types by default afaik also the scraems at me a lot otherwise for not leaving `kotlin.` away, so it seems fine.

Reply:
Just reverted it for now it shouldn't make a difference except generating some more warnings because of overqualified paths